source/Playlist.cpp

```diff
@@ -0,0 +1,146 @@
+/* Playlist.cpp
+Copyright (c) 2022 by RisingLeaf
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "Playlist.h"
+
+#include "DataNode.h"
+#include "GameData.h"
+#include "PlayerInfo.h"
+#include "System.h"
+
+#include <algorithm>
+
+using namespace std;
+
+namespace {
+	const Track *currentTrack = nullptr;
```

Comment:
Why store this in the namespace instead of in the header?

Reply:
as a static?

Reply:
As of right now it's in the namespace since this is the only place that would reference it.

Do you want this moved into header to proceed?